eng/pipelines/templates/steps/build-and-test.yml

```diff
@@ -50,7 +50,7 @@ steps:
       goals: 'install'
     env:
       AZURE_VERSION_OVERRIDE_TESTS: ${{ parameters.TestVersionSupport }}
-    condition: and(succeeded(), ne(variables['TestFromSource'], 'true'))
+    condition: or(and(succeeded(), ne(variables['TestFromSource'], 'true')), eq(variables['IsLatestNonLtsJdk'], true))
```

Comment:
I'm confused about the intent behind the `or` here. You want to run this step even if other steps fail when non latest is true?

Reply:
I think this can be changed thinking about it now as the extra linting steps being ran in the latest JDK run don't cause the pipeline to fail.